### ISSUE TITLE
bigchaindb transaction using js driver from Android Java app failure

### ISSUE THREAD

**Issue by adonnini** [bigchaindb/js-bigchaindb-driver#171]
Hi,

I was finally able to run the bigchaindb-js driver from my Android  Java application.

When running this code:

                    startNodeWithArguments(new String[]{"node", "-e",
                            "const driver = require('/data/user/0/com.yourorg.sample/files/bigchaindb-js/node_modules/bigchaindb-driver')\n" +
                                    "const API_PATH = 'https://test.ipdb.io/api/v2/'\n" +
                                    "const alice = new driver.Ed25519Keypair()\n" +
                                    "const tx = driver.Transaction.makeCreateTransaction({ city: 'Berlin, DE', temperature: 22, datetime: new Date().toString() }, { what: 'My first BigchainDB transaction' }, [ driver.Transaction.makeOutput(driver.Transaction.makeEd25519Condition(alice.publicKey))], alice.publicKey)\n" +
                                    "const txSigned = driver.Transaction.signTransaction(tx, alice.privateKey)\n" +
                                    "const conn = new driver.Connection(API_PATH)\n" +
                                    "conn.postTransactionCommit(txSigned)\n" +
                                    "    .then(retrievedTx => console.log('Transaction', retrievedTx.id, 'successfully posted.'))\n"                    });

The app failed with the following error log:

05-09 09:19:26.433: E/NODEJS-MOBILE(11278): { FetchError: request to https://test.ipdb.io/api/v2/transactions?mode=commit failed, reason: getaddrinfo ENOTFOUND test.ipdb.io test.ipdb.io:443
05-09 09:19:26.433: E/NODEJS-MOBILE(11278):     at ClientRequest.<anonymous> (/data/data/com.yourorg.sample/files/bigchaindb-js/node_modules/node-fetch/lib/index.js:1393:11)
05-09 09:19:26.433: E/NODEJS-MOBILE(11278):     at emitOne (events.js:115:13)
05-09 09:19:26.433: E/NODEJS-MOBILE(11278):     at ClientRequest.emit (events.js:210:7)
05-09 09:19:26.433: E/NODEJS-MOBILE(11278):     at TLSSocket.socketErrorListener (_http_client.js:401:9)
05-09 09:19:26.433: E/NODEJS-MOBILE(11278):     at emitOne (events.js:115:13)
05-09 09:19:26.433: E/NODEJS-MOBILE(11278):     at TLSSocket.emit (events.js:210:7)
05-09 09:19:26.433: E/NODEJS-MOBILE(11278):     at emitErrorNT (internal/streams/destroy.js:64:8)
05-09 09:19:26.433: E/NODEJS-MOBILE(11278):     at _combinedTickCallback (internal/process/next_tick.js:138:11)
05-09 09:19:26.433: E/NODEJS-MOBILE(11278):     at process._tickCallback (internal/process/next_tick.js:180:9)
05-09 09:19:26.433: E/NODEJS-MOBILE(11278):   message: 'request to https://test.ipdb.io/api/v2/transactions?mode=commit failed, reason: getaddrinfo ENOTFOUND test.ipdb.io test.ipdb.io:443',
05-09 09:19:26.433: E/NODEJS-MOBILE(11278):   type: 'system',
05-09 09:19:26.433: E/NODEJS-MOBILE(11278):   errno: 'ENOTFOUND',
05-09 09:19:26.433: E/NODEJS-MOBILE(11278):   code: 'ENOTFOUND' }

Am I using the wrong URL for the test network? What am I doing wrong?

Thanks,

Alex Donnini

**Comment by kansi** [comment]
@adonnini The URI has been migrated i.e. `https://test.ipdb.io` -> `https://test.bigchaindb.com`.
Other nodes are also available:  `https://test2.bigchaindb.com`,  `https://test3.bigchaindb.com` , `https://test4.bigchaindb.com`

**Comment by adonnini** [comment]
Does the above error have anything to do with incorrect authentication?

I have been wondering where to add my app_key and app_id,

Thanks,

Alex Donnini

**Comment by kremalicious** [comment]
@adonnini yes, the error suggests you are missing authentication. You can take a look at one of our guides to see how to pass the `app_id` & `app_key`:

- https://www.bigchaindb.com/developers/guide/tutorial-car-telemetry-app/#setup
- or see our Get Started page in the Node.js example code: https://www.bigchaindb.com/developers/getstarted/

tl;dr for your case:
```js
const conn = new driver.Connection(API_PATH, {
    app_id: 'Get one from testnet.bigchaindb.com',
    app_key: 'Get one from testnet.bigchaindb.com'
})
```

**Comment by adonnini** [comment]
@kremalicious Thanks! Hurdle overcome.

Next hurdle:

05-09 16:04:25.406: E/NODEJS-MOBILE(24674): { message: 'HTTP Error: Requested page not reachable',
05-09 16:04:25.406: E/NODEJS-MOBILE(24674):   status: '404 NOT FOUND',
05-09 16:04:25.406: E/NODEJS-MOBILE(24674):   requestURI: 'https://test.bigchaindb.com/api/v2/transactions?mode=commit' }

Is this because the test network I try to access (https://test.bigchaindb.com) is down? Should I try another?

Thanks,

Alex Donnini

**Comment by kremalicious** [comment]
Sorry, didn't see that before but the correct API path in your example should have a `v1` at the end, not v2. So like that:

```js
const API_PATH = 'https://test.bigchaindb.com/api/v1/'
```

**Comment by adonnini** [comment]
@kremalicious No problem. Now:

05-09 16:33:15.288: E/NODEJS-MOBILE(25683): { message: 'HTTP Error: Requested page not reachable',
05-09 16:33:15.288: E/NODEJS-MOBILE(25683):   status: '500 Internal Server Error',
05-09 16:33:15.288: E/NODEJS-MOBILE(25683):   requestURI: 'https://test.bigchaindb.com/api/v1/transactions?mode=commit' }

However, when I changed to the test2 network, success!

05-09 16:36:35.469: I/NODEJS-MOBILE(25966): Transaction 2266aa02c7ad1ff24dc27ba8bc7916d3da06458ea4f249e83d1d56ef259da61b successfully posted.

Thanks for all your help.

Alex Donnini
